### DISCUSSION TITLE
How to draw basic shapes?

### DISCUSSION THREAD

**Discussion by ghost** [cnjinhao/nana#707] [General]
It seems the built-in support for this is being deprecated. I got the deprecated message when I build the branch develop-1.8. Is there any alternatives planned? Or a third-party library like cairo must be used?

**Comment by qPCR4vir** [comment]
Hi...
yes, every widget will have now two new functions to registre a drawing function:
```
		std::function<void(paint::graphics&)> drawing() const;
		void drawing(std::function<void(paint::graphics&)>);
```
Before you had:
`	nana::drawing{ *this }.draw([this](nana::paint::graphics& graph) {		_m_render(graph);		});`
you constructed a drawing object, and registered there a function that use a graphic to draw.

Now you use:
`	this->drawing([this](nana::paint::graphics& graph) { 				_m_render(graph); 			});`

Were in this case `this `is the widget were you want to draw using the function `_m_render(graph)` which in turn draws by using the member functions in the class `graph`:. https://github.com/cnjinhao/nana/blob/8d79df06b25a55e127ca7e76943d8ef3a83f1ef4/include/nana/paint/graphics.hpp#L105
